examples/access-control-migration/src/services/user.service.ts

```diff
@@ -0,0 +1,57 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/example-access-control-migration
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {HttpErrors} from '@loopback/rest';
+import {UserRepository} from '../repositories/user.repository';
+import {User} from '../models/user.model';
+import {UserService} from '@loopback/authentication';
+import {UserProfile, securityId} from '@loopback/security';
+import {repository} from '@loopback/repository';
+
+export type Credentials = {
+  email: string;
+  password: string;
+};
+
+export class MyUserService implements UserService<User, Credentials> {
+  constructor(
+    @repository(UserRepository) public userRepository: UserRepository,
+  ) {}
+
+  async verifyCredentials(credentials: Credentials): Promise<User> {
+    const invalidCredentialsError = 'Invalid email or password.';
+
+    const foundUser = await this.userRepository.findOne({
+      where: {email: credentials.email},
+    });
+    if (!foundUser) {
+      throw new HttpErrors.Unauthorized(invalidCredentialsError);
+    }
+
+    const credentialsFound = await this.userRepository.findCredentials(
+      foundUser.id,
+    );
+    if (!credentialsFound) {
+      throw new HttpErrors.Unauthorized(invalidCredentialsError);
+    }
+    // TBD: Hash password
+    const passwordMatched = credentialsFound.password === credentials.password;
```

Comment:
Uh oh, storing plaintext passwords looks like a big red flag to me. Please ensure that hashing is implemented before this pull request lands.

Reply:
oops one more refactor, almost forgot it.

Reply:
added in the new commit